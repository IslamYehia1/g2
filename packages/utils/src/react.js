import React from 'react';
import { is } from './is';

export function getValidChildren(children) {
	return React.Children.toArray(children).filter((child) =>
		React.isValidElement(child),
	);
}

export function assignRef(ref, value) {
	if (ref == null) return;

	if (is.function(ref)) {
		ref(value);
		return;
	}

	try {
		//@ts-ignore
		ref.current = value;
	} catch (error) {
		throw new Error(`Cannot assign value '${value}' to ref '${ref}'`);
	}
}

export function mergeRefs(...refs) {
	return (value) => {
		refs.forEach((ref) => assignRef(ref, value));
	};
}
