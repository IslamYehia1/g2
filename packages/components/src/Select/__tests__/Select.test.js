import { render } from '@testing-library/react';
import React from 'react';

import { Select } from '../index';

describe('props', () => {
	const options = [
		{ value: 'olaf', label: 'Olaf' },
		{ value: 'samantha', label: 'Samantha' },
	];

	test('should render correctly', () => {
		const { container } = render(<Select id="select" options={options} />);
		expect(container.firstChild).toMatchSnapshot();
	});

	test('should render defaultValue', () => {
		const { container } = render(
			<Select defaultValue="olaf" id="select" options={options} />,
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	test('should render size', () => {
		const { container } = render(
			<Select id="select" options={options} size="small" />,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
