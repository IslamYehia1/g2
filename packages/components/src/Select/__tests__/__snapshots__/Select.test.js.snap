// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render children 1`] = `
<div
  class="wp-components-flex wp-components-base-field wp-components-select css-kmdofd"
  data-g2-c16t="true"
  data-g2-component="SelectWrapper"
>
  <select
    class="css-1vhtx5r"
    data-g2-c16t="true"
    data-g2-component="Select"
    id="select"
  >
    <optgroup
      label="Snowmen"
    >
      <option
        value="olaf"
      >
        Olaf
      </option>
    </optgroup>
  </select>
  <div
    class="css-2av5m8"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <span
      class="wp-components-truncate wp-components-text css-ckswz2"
      data-g2-c16t="true"
      data-g2-component="Text"
    >
      <div
        class="wp-components-icon css-1tw4gp0"
        data-g2-c16t="true"
        data-g2-component="Icon"
      >
        <svg
          fill="none"
          height="14"
          size="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="6 9 12 15 18 9"
          />
        </svg>
      </div>
    </span>
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
<div
  class="wp-components-flex wp-components-base-field wp-components-select css-kmdofd"
  data-g2-c16t="true"
  data-g2-component="SelectWrapper"
>
  <select
    class="css-1vhtx5r"
    data-g2-c16t="true"
    data-g2-component="Select"
    id="select"
  >
    <option
      value="olaf"
    >
      Olaf
    </option>
    <option
      value="samantha"
    >
      Samantha
    </option>
  </select>
  <div
    class="css-2av5m8"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <span
      class="wp-components-truncate wp-components-text css-ckswz2"
      data-g2-c16t="true"
      data-g2-component="Text"
    >
      <div
        class="wp-components-icon css-1tw4gp0"
        data-g2-c16t="true"
        data-g2-component="Icon"
      >
        <svg
          fill="none"
          height="14"
          size="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="6 9 12 15 18 9"
          />
        </svg>
      </div>
    </span>
  </div>
</div>
`;

exports[`props should render disabled 1`] = `
<div
  class="wp-components-flex wp-components-base-field wp-components-select css-kmdofd"
  data-g2-c16t="true"
  data-g2-component="SelectWrapper"
  disabled=""
>
  <select
    class="css-1vhtx5r"
    data-g2-c16t="true"
    data-g2-component="Select"
    disabled=""
    id="select"
  >
    <option
      value="olaf"
    >
      Olaf
    </option>
    <option
      value="samantha"
    >
      Samantha
    </option>
  </select>
  <div
    class="css-2av5m8"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <span
      class="wp-components-truncate wp-components-text css-ckswz2"
      data-g2-c16t="true"
      data-g2-component="Text"
    >
      <div
        class="wp-components-icon css-1tw4gp0"
        data-g2-c16t="true"
        data-g2-component="Icon"
      >
        <svg
          fill="none"
          height="14"
          size="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="6 9 12 15 18 9"
          />
        </svg>
      </div>
    </span>
  </div>
</div>
`;

exports[`props should render readOnly 1`] = `
<div
  class="wp-components-flex wp-components-base-field wp-components-select css-kmdofd"
  data-g2-c16t="true"
  data-g2-component="SelectWrapper"
>
  <select
    class="css-1vhtx5r"
    data-g2-c16t="true"
    data-g2-component="Select"
    id="select"
    readonly=""
  >
    <option
      value="olaf"
    >
      Olaf
    </option>
    <option
      value="samantha"
    >
      Samantha
    </option>
  </select>
  <div
    class="css-2av5m8"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <span
      class="wp-components-truncate wp-components-text css-ckswz2"
      data-g2-c16t="true"
      data-g2-component="Text"
    >
      <div
        class="wp-components-icon css-1tw4gp0"
        data-g2-c16t="true"
        data-g2-component="Icon"
      >
        <svg
          fill="none"
          height="14"
          size="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="6 9 12 15 18 9"
          />
        </svg>
      </div>
    </span>
  </div>
</div>
`;

exports[`props should render required 1`] = `
<div
  class="wp-components-flex wp-components-base-field wp-components-select css-kmdofd"
  data-g2-c16t="true"
  data-g2-component="SelectWrapper"
>
  <select
    class="css-1vhtx5r"
    data-g2-c16t="true"
    data-g2-component="Select"
    id="select"
    required=""
  >
    <option
      value="olaf"
    >
      Olaf
    </option>
    <option
      value="samantha"
    >
      Samantha
    </option>
  </select>
  <div
    class="css-2av5m8"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <span
      class="wp-components-truncate wp-components-text css-ckswz2"
      data-g2-c16t="true"
      data-g2-component="Text"
    >
      <div
        class="wp-components-icon css-1tw4gp0"
        data-g2-c16t="true"
        data-g2-component="Icon"
      >
        <svg
          fill="none"
          height="14"
          size="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="6 9 12 15 18 9"
          />
        </svg>
      </div>
    </span>
  </div>
</div>
`;

exports[`props should render size 1`] = `
<div
  class="wp-components-flex wp-components-base-field wp-components-select css-kmdofd"
  data-g2-c16t="true"
  data-g2-component="SelectWrapper"
>
  <select
    class="css-btkg83"
    data-g2-c16t="true"
    data-g2-component="Select"
    id="select"
  >
    <option
      value="olaf"
    >
      Olaf
    </option>
    <option
      value="samantha"
    >
      Samantha
    </option>
  </select>
  <div
    class="css-2av5m8"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <span
      class="wp-components-truncate wp-components-text css-ckswz2"
      data-g2-c16t="true"
      data-g2-component="Text"
    >
      <div
        class="wp-components-icon css-1tw4gp0"
        data-g2-c16t="true"
        data-g2-component="Icon"
      >
        <svg
          fill="none"
          height="14"
          size="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="6 9 12 15 18 9"
          />
        </svg>
      </div>
    </span>
  </div>
</div>
`;
